src/suggestion.c

```diff
@@ -0,0 +1,448 @@
+/*
+ * A series of functions to generate spelling suggestions
+ * 
+ * See suggestion.h for function documentation
+ */
+
+#include <stdlib.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <assert.h>
+#include <unistd.h>
+#include <string.h>
+#include "suggestion.h"
+#include "trie.h"
+
+int has_children(trie_t *t, char *s) {
+    
+    return !trie_search(t, s);
+}
+
+/* See suggestion.h */
+int cmp_match(const void* a, const void* b) {
+
+    // Recast a and b as match_t's
+    match_t *aa = *(match_t **)a;
+    match_t *bb = *(match_t **)b;
+
+    // Put null entries at the end of the list
+    if (aa == NULL) {
+        return 1;
+    } else if (bb == NULL) {
+        return -1;
+    }
+
+    // Prioritize matches with the higher edits_left score
+    if (aa->edits_left != bb->edits_left) {
+        return bb->edits_left - aa->edits_left;
+    }
+
+    // Otherwise sort lexographically
+    return strcmp(aa->str,bb->str);
+}
+
+// Helper function for suggestions that just moves on to the next character
+int move_on(match_t **set, trie_t *t, char *prefix, char *suffix, int edits_left, int n) {
+
+    int rc = EXIT_SUCCESS;
+    int len = strlen(prefix);
+    char* new_prefix;
+
+    new_prefix = malloc(sizeof(char) * (MAXLEN + 1));
+    if (new_prefix == NULL) {
+        return EXIT_FAILURE;
+    }
+
+    strncpy(new_prefix, prefix, MAXLEN);
+
+    new_prefix[len] = suffix[0];
+
+    if (has_children(t, new_prefix) == EXIT_SUCCESS) {
+
+        // Move on to the next character, don't use up an edit
+        rc = suggestions(set, t, new_prefix, suffix + 1, edits_left, n);
+    }
+
+    // Save some space now that we're done
+    free(new_prefix);
+
+    if (rc != EXIT_SUCCESS) {
```

Comment:
Why not just `return rc`?

Reply:
@dwahme

Reply:
Changed.

Reply:
@dwahme hmm, it doesn't look like this was changed as of the latest commits on `dustin/Spellcheck`...